XSOAR RF 2.3 release

rderkachrf opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues

## Description
In this release, we added the ability to pull alerts from the Recorded Future and view them as XSOAR Incidents. Two more commands were added to interact with Recorded Future alerts(set status and write note) 

## Screenshots
![image](https://user-images.githubusercontent.com/86717829/132469444-3b2f08d8-0c31-43a6-ba5a-e26d26e4f651.png)
![image](https://user-images.githubusercontent.com/86717829/132469500-520e63f0-51b6-43d9-9bb3-3ab339a365f7.png)
![image](https://user-images.githubusercontent.com/86717829/132469550-97f811e4-4a56-4fa9-8152-bf00a73810ad.png)
![image](https://user-images.githubusercontent.com/86717829/132469683-f7f5de44-eb91-4ab8-b38d-610d315b8d4c.png)
![image](https://user-images.githubusercontent.com/86717829/132469709-35615e7f-5e07-4c46-81d6-ad365c004277.png)


## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

ShahafBenYakir commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.yml:
```diff
@@ -2048,6 +2067,71 @@ script:
       description: Recorded Future Link Entity Risk Score
       type: Number
     description: Get Insikt Group Research Links for an IP, Domain, CVE, URL or File.
+  - name: recordedfuture-alert-writestatus
```
```suggestion
  - name: recordedfuture-alert-set-status
```

ShahafBenYakir commented on Packs/RecordedFuture/ReleaseNotes/1_1_5.md:
```diff
@@ -0,0 +1,6 @@
+##### Recorded Future v2
+- Updated the Docker image to: *demisto/python3:3.9.6.24067*.
+- Added support for fetching incidents using fetch_incidents method
+- Added two new commands `recordedfuture-alert-writestatus` and recordedfuture-alert-setnote`
```
```suggestion
- Added two new commands ***recordedfuture-alert-set-status*** and ***recordedfuture-alert-set-note***
```

ShahafBenYakir commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.yml:
```diff
@@ -2048,6 +2067,71 @@ script:
+      description: Recorded Future alert get date
+      type: date
+    description: Set alert into predefined status
+  - name: recordedfuture-alert-setnote
```
```suggestion
  - name: recordedfuture-alert-set-note
```

barchen1 commented on Packs/RecordedFuture/ReleaseNotes/1_1_5.md:
```diff
@@ -0,0 +1,6 @@
+##### Recorded Future v2
+- Updated the Docker image to: *demisto/python3:3.9.7.24076*.
+- Added support for fetching incidents using fetch_incidents method
+- Added two new commands `recordedfuture-alert-set-status` and recordedfuture-alert-set-note`
```
```suggestion
- Added two new commands ***recordedfuture-alert-set-status*** and ***recordedfuture-alert-set-note***```

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.yml:
```diff
@@ -63,11 +63,30 @@ configuration:
+- defaultvalue: 24 hours
+  display: First fetch time (<number> <time unit>, e.g., 12 hours, 7 days, 3 months,
+    1 year)
+  name: triggered
```
```suggestion
  name: fetch_time
```

rderkachrf replied:
demisto validator saying to set this to `first_fetch`
and found this doc
https://xsoar.pan.dev/docs/tutorials/tut-design#fetch-incidents-parameters
so set the name to  `first_fetch`

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.yml:
```diff
@@ -63,11 +63,30 @@ configuration:
+- display: Incident type
+  name: incidentType
+  required: false
+  type: 13
```
I think you should add the `max_fetch` parameter so customers can set the max results to fetch.
WDYT?

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.yml:
```diff
@@ -2048,6 +2067,71 @@ script:
+      description: Recorded Future alert get date
+      type: date
+    description: Set a note for the alert in Recorded Future
   isfetch: false
```
```suggestion
  isfetch: true
```

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture_test.py:
```diff
@@ -181,6 +181,30 @@ def test_get_links_command(self):
         self.assertTrue(context.get('HumanReadable'))
         self.assertTrue(context.get('Contents'))
 
+    @vcr.use_cassette()
```
Please add unit tests for the `fetch-incidents` command.

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:
```diff
@@ -1357,6 +1371,122 @@ def __build_links_context(self, links_data: Dict[str, List], markdown: str) -> C
 
+    def fetch_incidents(self, rule_names_arg: Optional[str], fetch_time: str) -> None:
+        if rule_names_arg:
+            rule_names = rule_names_arg.split(';')
```
Why split by `;` ?

rderkachrf replied:
That is because we specify that in the description for rule names, and that is the format we want to get.
```
-display: Rule names to fetch alerts by, separated by semicolon. If empty, all alerts
    will be fetched
 ```

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:
```diff
@@ -1382,20 +1512,25 @@ def main() -> None:
             return_results(actions.lookup_command(entities, command))
+        elif command == 'fetch-incidents':
+            rule_names = demisto_params.get('rule_names', '').strip()
+            fetch_time = demisto_params.get('triggered', '').strip()
```
```suggestion
            fetch_time = demisto_params.get('fetch_time', '24 hours').strip()
```

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:
```diff
@@ -1357,6 +1371,122 @@ def __build_links_context(self, links_data: Dict[str, List], markdown: str) -> C
+        else:
+            rule_names = []
+
+        if not fetch_time:
+            fetch_time = '24 hours'
```
You can remove these lines.
The default value for `fetch_time` is 24 hours.

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:
```diff
@@ -1357,6 +1371,122 @@ def __build_links_context(self, links_data: Dict[str, List], markdown: str) -> C
+
+        for rule in rule_names:
+            rules = self.client.get_alert_rules(rule)
+            if rules and 'data' in rules:
```
Not sure I understand this if statement.
Can you please explain?

rderkachrf replied:
there might be a response but with `error` key and no `data` key

barchen1 replied:
Thanks.
Is `rules` a string/dictionary ?

rderkachrf replied:
it's a dictionary.

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:
```diff
@@ -1357,6 +1371,122 @@ def __build_links_context(self, links_data: Dict[str, List], markdown: str) -> C
+            alert_time = datetime.strptime(alert['triggered'], '%Y-%m-%dT%H:%M:%S.%fZ')
+            # The API returns also alerts that are triggered in the same time
+            if alert_time > current_time:
+                #Set alerts status to pending
+                update_data.append({"id": alert['id'], "status": "pending"})
```
Why set the alert status to pending?

rderkachrf replied:
That is how we want to handle alerts that have been pulled from our system.  Pending means that we are still waiting for users to interact with it.

barchen1 commented on Packs/RecordedFuture/Integrations/RecordedFuture/README.md:
```diff
@@ -708,6 +708,116 @@ Get details on alerts configured and generated by Recorded Future by alert rule
 >| Global Trends, Trending Targets - Spike: Enel SPA, Knoxville and Alabama |
 
 
+### recordedfuture-alert-set-status
```
Please add a note fin the README file for the `fetch-incidents` command you added.

barchen1 reviewed (changes requested):
Thank you for your contribution!
Please go over the notes and let me know if you need my help.

bakatzir commented on Packs/RecordedFuture/ReleaseNotes/1_1_6.md:
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Recorded Future v2
+- Added support for fetching incidents using fetch_incidents method
+- Added two new commands ***recordedfuture-alert-set-status*** and ***recordedfuture-alert-set-note***
```
```suggestion
- Added support for fetching incidents.
- Added 2 new commands: ***recordedfuture-alert-set-status***, ***recordedfuture-alert-set-note***.
```

rderkachrf commented on Packs/RecordedFuture/.secrets-ignore:
```diff
@@ -51,3 +51,7 @@
+https://discord.com
+https://hackearonline.net
+https://www.watson.ch
+muzamilofficially@gmail.com
```
It's not a problem, but I will remove it.

rderkachrf commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:
```diff
@@ -1357,6 +1371,127 @@ def __build_links_context(self, links_data: Dict[str, List], markdown: str) -> C
+            time = last_run['time']
+
+        current_time = datetime.strptime(time, '%Y-%m-%dT%H:%M:%S.%fZ')
+        triggered_time = '[{},)'.format(datetime.strftime(current_time, '%Y-%m-%d %H:%M:%S'))
```
Yes. That is the custom time format for our API.

rderkachrf commented on Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture_test.py:
```diff
@@ -239,3 +265,71 @@ def test_entity_enrich_no_related_entities(mocker):
+def test_fetch_incidents(mocker):
+    """Fetch alerts from Recorded Future"""
+    first_fetch = "72 hours"
+    rule_names = "Global Trends, Trending Vulnerabilities;Global Trends, Trending Attackers"
```
It should.  And it is.
Global Trends, Trending Vulnerabilities - > is first
Global Trends, Trending Attackers -> second

rderkachrf commented on Packs/RecordedFuture/pack_metadata.json:
```diff
@@ -2,7 +2,7 @@
     "description": "Recorded Future App v2",
     "support": "partner",
-    "currentVersion": "1.1.5",
+    "currentVersion": "1.1.6",
```
Ok. I will make it 1.2.0 then

barchen1 commented:
@rderkachrf As we talked on our call, you should add incident-type, layout, mapper (if needed) cause you added the `fetch-incidents` command.
Let me know if you need help, you can also reach out to @edik24 or @kgal-pan for more guidance.

barchen1 commented on Packs/RecordedFuture/Layouts/layoutscontainer-RecordedFutureLayouts.json:
```diff
@@ -0,0 +1,316 @@
+    "system": false,
+    "version": -1,
+    "fromVersion": "6.0.0",
+    "description": ""
```
Please add description

barchen1 commented on Packs/RecordedFuture/IncidentTypes/customIncidentTypes.json:
```diff
@@ -0,0 +1,42 @@
+[
+	{
+		"id": "RecordedFuture",
```
I think we should think of a better name here
@evisochek What do you think?

evisochek replied:
I thinks its worth to add couple of words for better description of the incident type

rderkachrf replied:
Change the file name
@rderkachrf

barchen1 commented on Packs/RecordedFuture/IncidentTypes/customIncidentTypes.json:
```diff
@@ -0,0 +1,42 @@
+[
+	{
+		"id": "RecordedFuture",
+		"version": 6,
```
```suggestion
		"version": -1,
```

evisochek commented on Packs/RecordedFuture/IncidentTypes/customIncidentTypes.json:
```diff
@@ -0,0 +1,42 @@
+		"name": "RecordedFuture",
+		"prevName": "RecordedFuture",
+		"color": "#0071cf",
+		"playbookId": "playbook0",
```
You are not using any playbook for you incident type?

bakatzir commented on Packs/RecordedFuture/ReleaseNotes/1_2_0.md:
```diff
@@ -0,0 +1,50 @@
+
+#### Layouts
+##### New: RecordedFutureLayout
+- Add Recorded Future Layout
+
+#### Indicator Fields
+##### RecordedFuture Risk rules
+- Formatting file
+
+##### RecordedFuture Threat assessment
+- Formatting file
+
+#### Playbooks
+##### Recorded Future CVE Intelligence
+- Formatting file
+
+##### Recorded Future CVE Reputation
+- Formatting file
+
+##### Recorded Future Domain Intelligence
+- Formatting file
+
+##### Recorded Future Domain Reputation
+- Formatting file
+
+##### Recorded Future File Intelligence
+- Formatting file
+
+##### Recorded Future File Reputation
+- Formatting file
+
+##### Recorded Future IP Intelligence
+- Formatting file
+
+##### Recorded Future IP Reputation
+- Formatting file
+
+##### Recorded Future Threat Assessment
```
```suggestion
- Add Recorded Future Layout

#### Indicator Fields
##### RecordedFuture Risk rules
- Maintenance and stability enhancements.

##### RecordedFuture Threat assessment
- Maintenance and stability enhancements.

#### Playbooks
##### Recorded Future CVE Intelligence
- Maintenance and stability enhancements.

##### Recorded Future CVE Reputation
- Maintenance and stability enhancements.

##### Recorded Future Domain Intelligence
- Maintenance and stability enhancements.

##### Recorded Future Domain Reputation
- Maintenance and stability enhancements.

##### Recorded Future File Intelligence
- Maintenance and stability enhancements.

##### Recorded Future File Reputation
- Maintenance and stability enhancements.

##### Recorded Future IP Intelligence
- Maintenance and stability enhancements.

##### Recorded Future IP Reputation
- Maintenance and stability enhancements.

##### Recorded Future Threat Assessment
```

bakatzir commented on Packs/RecordedFuture/ReleaseNotes/1_2_0.md:
```diff
@@ -0,0 +1,50 @@
+- Formatting file
+
+##### Recorded Future Threat Assessment
+- Formatting file
+
+##### Recorded Future URL Intelligence
+- Formatting file
+
+##### Recorded Future URL Reputation
+- Formatting file
```
```suggestion
- Maintenance and stability enhancements.

##### Recorded Future URL Intelligence
- Maintenance and stability enhancements.

##### Recorded Future URL Reputation
- Maintenance and stability enhancements.
```

rderkachrf commented:
@barchen1 updated the Incident type as discussed:
1. Changed the name of the Recorded Future Incident Type to be more specific.
2. Changed the file name.
